src/dataflow/render/mod.rs

```diff
@@ -251,27 +253,21 @@ pub(crate) fn build_dataflow<A: Allocate>(
                         _ => false,
                     };
 
-                    let source_config = SourceConfig::new(
-                        uid,
-                        region,
-                        timestamp_histories.clone(),
-                        timestamp_channel.clone(),
+                    let source_config = SourceConfig {
+                        name: format!("{}-{}", connector.name(), uid),
```

Comment:
Right, it uses "uid" instead, which is a "unique ID" for the source. There's a comment on the definition of the variable that says "this must be unique across all instantiations of a source." Does that resolve your immediate concern? I'm trying to chip away at this code piece by piece.

Reply:
I think that if this comment is true it is only because of an uncommented early return back in `coord.rs` (my lines 1497-1501) that doesn't explain that if it gets commented out the system quietly fail.

I'd personally really much rather have an assert to check the invariant. Just one line saying that all source ids are unique, as the other part of the uid seems to be fixed per-dataflow desc.

Reply:
Alternately, we could pro-actively deduplicate `dataflow.source_imports`

Reply:
But it's already de-duped by the coordinator, right? I've added the assert!

Reply:
I hope that continues to be true! We have enough crates with public surfaces that I'd to rely less on details of how one crate happens to be used by other (because that breaks, and if it breaks it's hard to find out why, what the root cause was, etc).